src/networking.c

```diff
@@ -2262,9 +2264,11 @@ int processInlineBuffer(client *c) {
 
     /* Setup argv array on client structure */
     if (argc) {
-        if (c->argv) zfree(c->argv);
-        c->argv_len = argc;
-        c->argv = zmalloc(sizeof(robj*)*c->argv_len);
+        if (argc > c->argv_len) {
+            zfree(c->argv);
+            c->argv = zmalloc(sizeof(robj*)*argc);
```

Comment:
we can even try realloc instead of free + malloc

Reply:
but if argv changes from very large to very small, realloc might be slower than free+malloc.

Reply:
add your suggestion to TODO.

Reply:
I agree, it might be the downside. Realloc() would be a bet that for most cases, we won't need to move the memory. e.g. 24 bytes argv is allocated from 32 bytes pool, expanding by one item would be quicker than free + malloc. It might be something to try if this is still in a critical path after your change.